sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,201 @@
+from sympy import Symbol
+from sympy.physics.vector import Vector, cross
+from sympy.physics.mechanics import (dynamicsymbols, Body, PinJoint,
+                                     SlidingJoint, CylindricalJoint, PlanarJoint,
+                                     SphericalJoint)
+
+
+def test_pin_joint():
+    parent = Body('parent')
+    child = Body('child')
+    l = Symbol('l')
+    pin_joint = PinJoint('pin_joint', parent, child, child_point_pos=(0, l, 0))
+
+    assert pin_joint.name == 'pin_joint'
+    assert pin_joint.parent == parent
+    assert pin_joint.child == child
+    assert pin_joint.parent_joint_vector == Vector(0)
+    assert pin_joint.child_joint_vector == l * child.frame.y
```

Comment:
Also these could be private since you have a Point object that contains this information.